agent-inject/agent/container_sidecar.go

```diff
@@ -151,12 +151,14 @@ func (a *Agent) createLifecycle() corev1.Lifecycle {
 
 func (a *Agent) securityContext() *corev1.SecurityContext {
 	runAsNonRoot := true
+
 	if a.RunAsUser == 0 || a.RunAsGroup == 0 {
 		runAsNonRoot = false
 	}
 	return &corev1.SecurityContext{
-		RunAsUser:    pointerutil.Int64Ptr(a.RunAsUser),
-		RunAsGroup:   pointerutil.Int64Ptr(a.RunAsGroup),
-		RunAsNonRoot: pointerutil.BoolPtr(runAsNonRoot),
+		RunAsUser:              pointerutil.Int64Ptr(a.RunAsUser),
+		RunAsGroup:             pointerutil.Int64Ptr(a.RunAsGroup),
+		RunAsNonRoot:           pointerutil.BoolPtr(runAsNonRoot),
+		ReadOnlyRootFilesystem: pointerutil.BoolPtr(true),
```

Comment:
Should we use `DefaultAgentReadOnlyRoot` here instead of `true`?

Reply:
Good catch, thought I got all of them.